app/src/services/objectPermission.js

```diff
@@ -61,6 +61,22 @@ const service = {
     }
   },
 
+  /**
+   * @function getObjectIdsWithBucket
+   * Searches for specific (object) bucket permissions
+   * @param {string|string[]} [params.userId] Optional string or array of uuids representing the user
+   * @param {string|string[]} [params.bucketId] Optional string or array of bucket id(s)
+   * @returns {Promise<object>} The result of running the find operation
+  */
+  getObjectIdsWithBucket: async (userId, bucketId) => {
+    return BucketPermission.query()
+      .distinct('object.id AS objectId')
+      .rightJoin('object', 'bucket_permission.bucketId', '=', 'object.bucketId')
+      .modify('filterUserId', userId)
+      .whereIn('bucket_permission.bucketId', bucketId)
```

Comment:
Double check but I think this could be a modify `filterBucketId` call.

Reply:
there is but it caused ambiguous ref to bucketId.

Reply:
Change the ordering of the statements so that they're logically grouped then:
```js
      .distinct('object.id AS objectId')
      .modify('filterUserId', userId)
      .rightJoin('object', 'bucket_permission.bucketId', '=', 'object.bucketId')
      .whereIn('bucket_permission.bucketId', bucketId)
```

Reply:
Also may be worth using `.rightJoinRelated` instead of rightJoin as it'll let us leverage the ORM relation mappings instead.
https://vincit.github.io/objection.js/api/query-builder/join-methods.html#rightjoinrelated